plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -1107,15 +1107,15 @@ functions.
 -}
 
 -- | Take a function and a list of arguments and apply the former to the latter.
-headSpine :: Opaque val asToB -> [val] -> Opaque (HeadSpine val) b
-headSpine (Opaque f) = Opaque . \case
+headSpineOpaque :: Opaque val asToB -> [val] -> Opaque (MonoHeadSpine val) b
+headSpineOpaque (Opaque f) = Opaque . \case
```

Comment:
Reuse `headSpine`? Doesn't matter much, I'm gonna remove this anyway.

Reply:
Yes, with the removal of `CaseData` and `CaseList`

Reply:
I think it's okay to leave this actually. This can be useful future for adding other builtins